driver-core/src/main/com/mongodb/internal/async/AsyncBatchCursor.java

```diff
@@ -72,4 +75,22 @@ public interface AsyncBatchCursor<T> extends Closeable {
      */
     @Override
     void close();
+
+    default void exhaustCursor(final SingleResultCallback<List<List<T>>> finalCallback) {
+        List<List<T>> results = new ArrayList<>();
+
+        beginAsync().thenRunDoWhileLoop(iterationCallback -> {
+                    beginAsync().
+                            thenSupply(this::next)
+                            .thenConsume((batch, callback) -> {
+                                if (batch != null && !batch.isEmpty()) {
+                                    results.add(batch);
+                                }
+                                callback.complete(callback);
+                            }).finish(iterationCallback);
+                }, () -> !this.isClosed())
+                .<List<List<T>>>thenSupply(callback -> {
+                    callback.complete(results);
+                }).finish(finalCallback);
```

Comment:
Let's use the formatting as recommended by `AsyncRunnable`. Also, the indentation seems off regardless.

```suggestion
        beginAsync().thenRunDoWhileLoop(iterationCallback -> {
            beginAsync().<List<T>>thenSupply(c -> {
                next(c);
            }).thenConsume((batch, c) -> {
                if (batch != null && !batch.isEmpty()) {
                    results.add(batch);
                }
                c.complete(c);
            }).finish(iterationCallback);
        }, () -> !this.isClosed()
        ).<List<List<T>>>thenSupply(c -> {
            c.complete(results);
        }).finish(finalCallback);
```